Game/Entities/bullet.py

```diff
@@ -58,3 +60,12 @@ def checkHit(self):
     def hit(self, damage: int):
         super().hit(damage)
         self.kill()
+
+    def kill(self, removeFromEntities=True):
+        """Kills the entity: Removes it from the currently active entities"""
+        if self.isAlive:
+            self.isAlive = False
+            self.currentSpeed = 0
+            self.gameState.animations.append(BulletExplosion(self.gameState, self.position))
+            if removeFromEntities:
+                self.gameState.entities.remove(self)
```

Comment:
Two suggestions here:
- Make use of `super().kill()` to avoid duplicated code
- Move the animation up by half a tile (this looks more accurately to be the position that the bullet hits)
```suggestion
    def kill(self, removeFromEntities=True):
        """Kills the bullet: Display an explosion animation"""
        if self.isAlive:
            self.gameState.animations.append(BulletExplosion(self.gameState,
                                                             self.position + Vector2(0, -self.gameState.tileSize.y/2)))
        super().kill(removeFromEntities)
```